lib/utils/cloud.ts

```diff
@@ -208,17 +209,18 @@ export async function downloadOSImage(
 async function resolveOSVersion(
 	deviceType: string,
 	version: string,
+	esr: boolean,
 ): Promise<string> {
 	if (['menu', 'menu-esr'].includes(version)) {
 		return await selectOSVersionFromMenu(deviceType, version === 'menu-esr');
 	}
+	// Note that `version` may also be 'latest', 'recommended', 'default'
 	if (/^v?\d+\.\d+\.\d+/.test(version)) {
 		if (version[0] === 'v') {
 			version = version.slice(1);
 		}
-		// The version must end with either '.dev' or '.prod', as expected
-		// by `balena-image-manager` and the balena SDK.
-		if (!version.endsWith('.dev') && !version.endsWith('.prod')) {
+		// ESR versions must end with either '.dev' or '.prod'
+		if (esr && !version.endsWith('.dev') && !version.endsWith('.prod')) {
 			version += '.prod';
```

Comment:
@thgreasi, when we last talked about it, you had a preference for updating CLI v12 (that uses SDK v15) to remove the `version += '.prod'` patch above, before releasing CLI v13 (that uses SDK v16).

But I think that was before we realised that balenaOS ESR versions are all invalid semvers and, without the `version += '.prod'` patch (CLI v12 + SDK v15), we get:

```
$ ./bin/balena-dev os download -o os.img --version 2021.10.2 raspberrypi4-64
Getting device operating system for raspberrypi4-64
BalenaRequestError: Request error: "Server error"
```

(Both staging and production.) After Alex G released 2.88.4+rev0 to staging, the patch is no longer needed for 2.88.4 though:

```
$ BALENARC_BALENA_URL=balena-staging.com ./bin/balena-dev os download -o os.img --version 2.88.4 raspberrypi4-64
Getting device operating system for raspberrypi4-64
balenaOS image version 2.88.4+rev0.prod downloaded successfully
```

So how about only appending `'.prod'` to ESR versions? This is what this PR is currently proposing, resulting in:

```
$ ./bin/balena-dev os download -o os.img --version 2021.10.2 raspberrypi4-64
Getting device operating system for raspberrypi4-64
balenaOS image version 2021.10.2.prod downloaded successfully
```

Do you expect that future balenaOS ESR versions (unified variants) would fail to download as a result of `version += '.prod'` here?

If you still think that it would be best to simply remove `version += '.prod'` from CLI v12 (regardless of ESR or not), I will do that.

Reply:
Further discussion in an [FD thread](https://www.flowdock.com/app/rulemotion/r-architecture/threads/YN4RClMOz6v6W-VCfoqBaNoliFZ).
We believe that appending `.prod` to the version would cause `balena os download` (CLI v12) to fail for future balenaOS ESR versions, and therefore we are removing the `version += '.prod'` patch.